src/socket.cr

```diff
@@ -244,13 +295,8 @@ class Socket < IO::FileDescriptor
   # ```
   def send(message)
     slice = message.to_slice
-    bytes_sent = LibC.send(fd, slice.to_unsafe.as(Void*), slice.size, 0)
-    raise Errno.new("Error sending datagram") if bytes_sent == -1
-    bytes_sent
-  ensure
-    # see IO::FileDescriptor#unbuffered_write
-    if (writers = @writers) && !writers.empty?
-      add_write_event
+    write_syscall_helper(slice, "Error sending message") do |slice|
```

Comment:
I realise that it's way beyond the scope of this PR but I really think we should have seperate types for datagram and stream-type sockets. The `send` syscall is basically overloaded to do 2 different things: copy data to the kernel's output buffer for stream sockets and send a message for datagram sockets. I think it's a mistake to confuse these by putting it all under one type.